test/all.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+. $DIR/common.sh
+
+for file in $(find "$DIR" -name "*.sh") ; do
```

Comment:
Motivation: it's a lot more convenient to have small files as they can be run individually.
I have already run into this before, e.g. https://github.com/dlang/installer/pull/297 (contains the same idea with `test/all.sh`)